pkg/adsclient/sotw/v3/client.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2022 Envoyproxy Authors
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+// Package sotw provides an implementation of GRPC SoTW (State of The World) part of XDS client
+package sotw
+
+import (
+	"context"
+	"errors"
+	"io"
+	"sync"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/golang/protobuf/ptypes/any"
+	status "google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	grpcStatus "google.golang.org/grpc/status"
+)
+
+var (
+	errInit    = errors.New("ads client: grpc connection is not initialized (use InitConnect() method to initialize connection)")
```

Comment:
Can we export these? Having users interact with errors using
```go
errors.Is(err, ErrInit)
```
is a pleasant user experience IMO.